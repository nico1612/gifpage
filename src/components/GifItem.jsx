import PropTypes from 'prop-types'

export const GiftItem=({title,url,id})=>{
    return(
        <div className="card">
            <img src={url} alt={url}/>
            <p>{title}</p>
        </div>
    )
}

CounterApp.propTypes={
    title:PropTypes.isRequired,
    url:PropTypes.isRequired
}