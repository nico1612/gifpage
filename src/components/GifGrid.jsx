import {GiftItem} from './GifItem'
import {useFetchGif} from '../hooks/useFetchGif'
import PropTypes from 'prop-types'

export const GifGrid=({category})=>{

    const { images, isLoading } = useFetchGif( category );

    return(
        <>
            <h3>{category}</h3>

            {
                isLoading && (<h2>cagando...</h2>)
            }
            <div className="card-grid">
            {
                images.map( ( image ) => (
                    <GiftItem 
                        key={ image.id } 
                        { ...image }
                    />
                ))
            }
            </div>
        </>
    )
}

GifGrid.propTypes={
    category:PropTypes.string.isRequired
}