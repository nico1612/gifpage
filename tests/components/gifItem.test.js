import { render,screen } from "@testing-library/react"
import { GiftItem } from "../../src/components/GifItem"


describe('pruebas en <GifItem.jsx />',()=>{
    const title ='saitama'
    const url ='http://one-punch.com/saitama.jpg'

    test('debe ser match con el snapshot',()=>{
        const {container}=render(<GiftItem title={title} url={url}/>)
        expect(container).toMatchSnapshot()
    })

    test('debe mostrar la imagen con el url y el alt indicado',()=>{
        render(<GiftItem title={title} url={url}/>)
        //expect(screen.getByRole('img').src).toBe(url);
        //expect(screen.getByRole('img').alt).toBe(title)
        const {src,alt}=screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    })

    test('debe mostrar el titulo en el componente',()=>{
        render(<GiftItem title={title} url={url}/>)
        expect(screen.getByText(title)).toBeTruthy()
    })
})