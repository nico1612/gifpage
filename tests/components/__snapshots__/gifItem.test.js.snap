// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifItem.jsx /> debe ser match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="saitama"
      src="http://one-punch.com/saitama.jpg"
    />
    <p>
      saitama
    </p>
  </div>
</div>
`;
